File: consistency-store/src/main/java/de/tudarmstadt/consistency/store/local/MapRef.java
package de.tudarmstadt.consistency.store.local;

import de.tudarmstadt.consistency.store.impl.ReadWriteRef;

/**
 * Created on 24.05.18.
 *
 * @author Mirko Köhler
 */
public class MapRef<T> extends ReadWriteRef<T> {

	private final Object id;
	private final MapStore store;


	MapRef(Object id, MapStore store) {
		this.id = id;
		this.store = store;
	}


	@Override
	protected T handleRead() throws Exception {
		return (T) store.get(id);
	}

	@Override
	protected void handleWrite(T value) throws Exception {
		store.put(id, value);
	}


}

File: consistency-store/src/main/scala/de/tudarmstadt/consistency/store/scala/local/MapRef.scala
package de.tudarmstadt.consistency.store.scala.local

import de.tudarmstadt.consistency.store.scala.impl.ReadWriteRef

import scala.collection.mutable

/**
		* References to values in the map.
 *
		* @param key the key that is referenced
		* @tparam T the value type that is refernced by this key
		*/
	class MapRef[Key, T](val data : mutable.Map[Key, Any], val key : Key) extends ReadWriteRef[Key, T] {

		override protected def handleRead(): Option[T] =
			data.get(key).asInstanceOf[Option[T]]

		override protected def handleWrite(t: T): Unit =
			data.put(key, t)
	}